cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -433,6 +443,18 @@ TEST(Grouper, NumericKey) {
 
     g.ExpectConsume("[[3], [27], [3], [27], [null], [81], [27], [81]]",
                     "[0, 1, 0, 1, 3, 2, 1, 2]");
+
+    g.ExpectFind("[[3], [3]]", "[0, 0]");
+
+    g.ExpectFind("[[3], [3]]", "[0, 0]");
```

Comment:
Did you mean to run an identical test twice?  Not sure if this is a copy/paste or you are testing for idempotence/deterministic behavior.

Reply:
I believe that was the intent. I was following the tests for `Consume` method. 
https://github.com/apache/arrow/blob/c9c97bd82534d91e68f444ad963521c01a39e213/cpp/src/arrow/compute/kernels/hash_aggregate_test.cc#L438-L440